src/dao/personDao.js

```diff
@@ -1,28 +1,50 @@
-var validation = require('../utils/validations');
-var personDB = require('../model/personModel.js');
+var personDB = require('../model/personModel');
 
-/*Validation constructor */
-function PersonDao() {
-}
-
-/*REturn messages by listMessagesId*/
-PersonDao.prototype.updateStateById = function (personId) {
+class PersonDao {
+    constructor() { }
+    
+    getPersonByEmail(personEmail) {
+        const getPersonByEmailPromise = new Promise((resolve, reject) => {
+            personDB.findOne({ 'email': personEmail }, function (err, personFound) {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(personFound);
+                }
+            });
+        });
+        return getPersonByEmailPromise;
+    };
 
-    personDB.findOne(
-        { '_id': personId },
-        '_id',
-        function (err, person) {
-            person.subscribed = "true";
-            person.save(function (err, createdTodoObject) {
+    getPersonById(personId) {
+        const getPersonByEmailPromise = new Promise((resolve, reject) => {
+            personDB.findOne({ '_id': personId }, function (err, personFound) {
                 if (err) {
-                    console.log(err);
-                    
+                    reject(err);
                 } else {
-                    console.log("Person updated");
-                    console.log(createdTodoObject);
+                    resolve(personFound);
                 }
             });
         });
-};
+        return getPersonByEmailPromise;
+    };
+
+    
+    updateStateById(personId) {
+        const updateStateByIdPromise = new Promise((resolve, reject) => {
+            personDB.findOne({ '_id': personId }, function (err, person) {
+                person.subscribed = "true";
```

Comment:
Why we store a boolean value inside a String? is not a good practice.

Reply:
Fixed